components/eamxx/src/share/io/scream_io_file_specs.hpp

```diff
@@ -56,10 +59,15 @@ struct StorageSpecs {
   }
 
   void update_storage (const util::TimeStamp& t) {
+    // We always update the snap counter (regardless of storage type),
```

Comment:
This was the bug: the method `needs_flush` checks the number of snapshots stored, so we must always update `num_snapshots_in_file` in order to handle flushing correctly for storage type != `NumSnaps`.